scripts/prodsec-manifest-generator.sh

```diff
@@ -1,82 +1,131 @@
 #!/usr/bin/env bash
 
-# This script generates a manifest of the integreatly-operator which includes the direct and indirect dependencies 
-# used. The generated manifest corresponds to a released version and is located in : ../rhoam-manifests/
-
 # Compares current manifest vs manifest generated for master
 manifest_compare() {
-    # Additional sorting of the master manifest is required as it seems that prow sorts files in a different manner.
-    sort -u "$RHOAM_CURRENT_MASTER" > "$RHOAM_SORTED_MASTER"
-
-    MANIFESTS_DIFF=$(diff --suppress-common-lines ${RHOAM_MASTER_FROM_BRANCH} ${RHOAM_SORTED_MASTER})
+    sort $MASTER_FROM_BRANCH > $SORTED_FROM_BRANCH
+    sort $CURRENT_MASTER > $SORTED_CURRENT_MASTER
+    MANIFESTS_DIFF=$(diff --suppress-common-lines ${SORTED_FROM_BRANCH} ${SORTED_CURRENT_MASTER})
     if [ ! -z "$MANIFESTS_DIFF" ]; then
-        echo "Difference found between master manifests, run `TYPE_OF_MANIFEST=master make manifest/prodsec` and push PR again"
-        
-        # Delete sorted files
-        rm -f "$RHOAM_MASTER_FROM_BRANCH"
-        rm -f "$RHOAM_SORTED_MASTER"
-        exit 1
-    else
+        echo "Difference found between master manifests"
+        rm -f "$MASTER_FROM_BRANCH"
+        rm -f "$SORTED_FROM_BRANCH"
+        rm -f "$SORTED_CURRENT_MASTER"
+        exit 0
+    elif [ -z "$MANIFESTS_DIFF" ]; then
         echo "No difference found between the manifests"
-
-        # Delete sorted files
-        rm -f "$RHOAM_MASTER_FROM_BRANCH"
-        rm -f "$RHOAM_SORTED_MASTER"
+        rm -f "$MASTER_FROM_BRANCH"
+        rm -f "$SORTED_FROM_BRANCH"
+        rm -f "$SORTED_CURRENT_MASTER"
         exit 0
+    else 
+        exit 1
     fi
 }
 
 # Generates the manifest, it can be either master or production manifest
 manifest_generate() {
-    # Pre-filetered manifest file
-    PRE_SORTED_FILE="prodsec-manifests/pre-sorted-file.txt"
+    FILEPATH="${CURRENT_DIR}/products/products.yaml"
+    PRODUCTS=10
+    # Generation manifest file
+    GENERATION_FILE="${CURRENT_DIR}/prodsec-manifests/generation-file.txt"
+
+    # Scan repos from the products.yaml file
+    for (( i=0; i<=$PRODUCTS; i++))
+    do
+        PRODUCT_NAME=$(yq r ${FILEPATH} "products[$i].name")
+        PRODUCT_VERSION=$(yq r ${FILEPATH} "products[$i].version")
+        PRODUCT_VERSION_ID=$PRODUCT_NAME:$PRODUCT_VERSION
+        PRODUCT_URL=$(yq r ${FILEPATH} "products[$i].url")
+        PRODUCT_INSTALL_TYPE=$(yq r ${FILEPATH} "products[$i].installType")
+
+        if [[ "$PRODUCT_INSTALL_TYPE" == *"$OLM_TYPE"* ]]; then
```

Comment:
We could get the values in the yaml file dynamically 

```suggestion
        if [[ "$(cat "${FILE}" | yq -r '.products[].installType')" == *"$OLM_TYPE"* ]]; then
```

Reply:
Thanks @jjaferson good point.